### PR TITLE
feature: Ability to silence alerts

### PR DESCRIPTION
**Pull request by dzsak**

### CODE REVIEW

**Comment by dzsak** [comment]
Screenshot 2023-12-15 at 14 38 49
Screenshot 2023-12-15 at 14 38 56

File: pkg/dashboard/notifications/message.go [hunk 1]
```diff
@@ -8,6 +8,7 @@ type Message interface {
 	RepositoryName() string
 	SHA() string
 	CustomChannel() string
+	Silenced() bool
```
Comment (laszlocph) [inline pkg/dashboard/notifications/message.go:11]: This belongs to the alert not the message.

**Review by laszlocph — CHANGES_REQUESTED** [review]

**Comment by laszlocph** [comment]
Please also show on the alert if it is silenced. Instead of the button, show a text: "Silenced until: xxx"

File: pkg/dashboard/alert/alertStateManager.go [hunk 2]
```diff
@@ -191,12 +195,14 @@ func (a AlertStateManager) TrackPod(pod *api.Pod, repoName string, envName strin
...
 			}
 
-			apiAlert := api.NewAlert(nonResolvedAlert, t.Text(), t.Name())
-			if previousState == model.FIRING { // don't notify people about pending then resolved alerts
-				a.notifManager.Broadcast(&notifications.AlertMessage{
-					Alert:       *apiAlert,
-					ImChannelId: pod.ImChannelId,
-				})
+			apiAlert := api.NewAlert(nonResolvedAlert, t.Text(), t.Name(), "")
+			if a.alertsSilenced(nonResolvedAlert.DeploymentName, nonResolvedAlert.Type) {
```
Comment (laszlocph) [inline pkg/dashboard/alert/alertStateManager.go:199]: This logically should be `if !a.alertsSilenced`

File: pkg/dashboard/notifications/alertMessage.go [hunk 3]
```diff
@@ -51,7 +51,7 @@ func (am *AlertMessage) AsSlackMessage() (*slackMessage, error) {
 							Type: button,
 							Text: &Text{
 								Type: "plain_text",
-								Text: "View",
+								Text: "Silence",
```
Comment (laszlocph) [inline pkg/dashboard/notifications/alertMessage.go:54]: There should be both a `View` and a `Silence` buttons.

File: pkg/dashboard/server/router.go [hunk 4]
```diff
@@ -182,6 +182,7 @@ func userRoutes(r *chi.Mux, clientHub *streaming.ClientHub) {
 		r.Post(("/api/environments"), saveInfrastructureComponents)
 		r.Post(("/api/bootstrapGitops"), bootstrapGitops)
 		r.Post(("/api/env/{env}/seal"), seal)
+		r.Post("/api/silenceAlerts", silenceAlerts)
```
Comment (laszlocph) [inline pkg/dashboard/server/router.go:185]: Should be `/api/silenceAlert`, since we silence a single alert at a time.

**Review by laszlocph — CHANGES_REQUESTED** [review]
Looks, good, just minor naming and semantics.

**Comment by dzsak** [comment]
Screenshot 2023-12-18 at 14 01 30

Screenshot 2023-12-18 at 14 01 37

**Comment by dzsak** [comment]
Screenshot 2023-12-18 at 19 01 02

**Review by laszlocph — APPROVED** [review]

**Pull request merged**